src/sonic-host-services/scripts/aaastatsd

```diff
@@ -0,0 +1,222 @@
+
+
+class RadiusStatistics:
+    def __init__(self, cfg_db, rad_global_conf, radius_conf):
+
+        self.radius_counter_names = [
+            "counter_0",
+            "access_requests",
+            "access_accepts",
+            "access_rejects",
+            "accounting_requests",
+            "accounting_responses",
+            "counter_6",
+            "counter_7",
+            "counter_8",
+            "counter_9",
+            "counter_10",
+            "access_challenges",
+            "counter_12",
+            "counter_13",
+            "counter_14",
+            "counter_15",
+            "counter_16",
+            "retried_access_requests",
+            "counter_18",
+            "counter_19",
+            "retried_accounting_requests",
+            "counter_21",
+            "counter_22",
+            "counter_23",
+            "counter_24",
+            "counter_25",
+            "counter_26",
+            "retried_access_challenges",
+            "counter_28",
+            "counter_29",
+            "counter_30",
+            "counter_31",
+            "timeouts",
+            "bad_authenticators",
+            "invalid_packets",
+            "counter_35",
+        ]
+
+        self.radius_global = {
+            'statistics': 'False'
+        }
+
+        self.radius_servers = {}
+
+        self.config_db = cfg_db
+
+        for row in rad_global_conf:
+            self.radius_global_update(row, rad_global_conf[row])
+
+        for row in radius_conf:
+            self.radius_server_update(row, radius_conf[row])
+
+
+        self.counters_db = ConfigDBConnector()
+        self.counters_db.db_connect('COUNTERS_DB', wait_for_init=False,
+		retry_on=True)
+        syslog.syslog(syslog.LOG_INFO, 'CountersDB connect success')
+        self.dbmon_thread = RadiusCountersDbMon("RadiusCountersDbMon",
+                                "RadiusCountersDbMon", self)
+        self.dbmon_thread.daemon = True
+        self.dbmon_thread.start()
+
+        self.filemon = RadiusStatsFileMon(self)
+        syslog.syslog(syslog.LOG_INFO, 'RADIUS Stats File Monitor started')
+
+    def radius_global_update(self, key, data):
+        if key == 'global':
+            self.radius_global.update(data)
+
+        for addr in self.radius_servers:
+            self.create_file(addr)
+
+    def radius_server_update(self, key, data):
+        if data == {}:
+            if key in self.radius_servers:
+                del self.radius_servers[key]
+        else:
+            self.radius_servers[key] = data
+
+        self.create_file(key)
+
+    def create_file(self, addr):
+        # print( "RadiusStatistics.create_file({})".format(addr))
+        stats_file = RADIUS_PAM_AUTH_CONF_STATS_DIR + addr
+        if self.radius_global['statistics'] == 'False':
+            if os.path.exists(stats_file):
+                os.unlink(stats_file)
+        else:
+            open(stats_file, 'a').close()
+            os.chmod(stats_file, 0o666)
```

Comment:
Semgrep detects a widely permissive file permission issue.
```
 scripts/aaastatsd
     python.lang.security.audit.insecure-file-permissions.insecure-file-permissions
        These permissions `0o666` are widely permissive and grant access to more people than may be
        necessary. A good default is `0o644` which gives read and write access to yourself and read
        access to everyone else.
        Details: https://sg.run/AXY4

        154┆ os.chmod(stats_file, 0o666)
```
Github code scanning also detects same issue.
Is it possible to limit the rw access to the owner only and give no access to others?

Reply:
+@a-barboza

Reply:
@maipbui 
I think 0o644 or 0o600 might be ok. There is no sensitive information in this file.

Reply:
Thanks @a-barboza! I’ll raise a PR to address this.